### PR TITLE
create_func_* now return strings

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

**Comment by mohamed-barakat** [comment]
@zickgraf: This is still using the legacy `CategoryConstructor`. It would be nice if `CAP`'s `CategoryConstructor` could be enhanced to simplify the code of the various `create_func_*`. The ultimate goal is of course compilation, but the readability of the `functor_category!.added_functions` is also important for me.

File: gap/FunctorCategories.gi [hunk 1]
```diff
@@ -821,8 +848,8 @@ InstallMethodWithCache( FunctorCategory,
                     fi;
                     
                     if not IsBound( images_of_objects[pos] ) then
-                        L := List( eval_arg, F -> ApplyCell( F, objB ) );
-                        images_of_objects[pos] := CallFuncList( oper, L );
+                        L := List( [ underlying_arguments ], cat_or_F_or_eta -> ApplyCell( cat_or_F_or_eta, objB ) );
```
Comment (mohamed-barakat) [inline gap/FunctorCategories.gi:851]: Locally deactivating caching by switching this line with the above `if`-line introduces a huge regression in `CatReps/examples/CategoryOfRepresentations.g`.

Comment (zickgraf) [reply gap/FunctorCategories.gi:851]: Would it make sense to add this finding as a comment in the code?

**Comment by zickgraf** [comment]
> @zickgraf: This is still using the legacy `CategoryConstructor`. It would be nice if `CAP`'s `CategoryConstructor` could be enhanced to simplify the code of the various `create_func_*`. The ultimate goal is of course compilation, but the readability of the `functor_category!.added_functions` is also important for me.

Is there a blocker why you cannot switch to `CAP`'s `CategoryConstructor`?

The first step for further simplification can already be done here: For example, you can compute the value of `functorial` as before and simply insert it in the string (e.g. by using `ReplacedStringViaRecord`). If we see that we insert the same strings in multiple packages, we can than put the corresponding logic in `CAP`'s `CategoryConstructor`.

File: gap/FunctorCategories.gi (hunk 1 above)
Comment (mohamed-barakat) [reply gap/FunctorCategories.gi:851]: Done with 1503b10e.

**Pull request merged**